keras2onnx/common/onnx_ops.py

```diff
@@ -112,28 +108,46 @@ def apply_cast(scope, input_name, output_name, container, operator_name=None, to
     d = onnx_proto.TensorProto.DataType.DESCRIPTOR
     allowed_type_name_and_type_enum_pairs = {v.number: k for k, v in d.values_by_name.items()}
     if to not in allowed_type_name_and_type_enum_pairs:
-        raise ValueError('Attribute to must be one of %s' % allowed_type_name_and_type_enum_pairs.keys())
+        raise ValueError('Attribute "to" must be one of %s' % allowed_type_name_and_type_enum_pairs.keys())
 
-    if container.target_opset < 7:
-        # Convert enum to string, for example, TensorProto.INT64 to 'INT64'
-        attrs['to'] = allowed_type_name_and_type_enum_pairs[to]
-        op_version = 1
+    if container.target_opset < 9:
+        if to in [onnx_proto.TensorProto.STRING, onnx_proto.TensorProto.COMPLEX64, onnx_proto.TensorProto.COMPLEX128]:
+            raise ValueError('Attribute "to" cannot correspond to a String or Complex TensorProto type.')
+
+        if container.target_opset < 6:
+            # Convert enum to string, for example, TensorProto.INT64 to 'INT64'
+            attrs['to'] = allowed_type_name_and_type_enum_pairs[to]
+            op_version = 1
+        else:
+            # Enum, for example, TensorProto.INT64
+            attrs['to'] = to
+            op_version = 6
     else:
-        # Enum, for example, TensorProto.INT64
+        # Enum value, for example, TensorProto.INT64
+        # String casting is supported in opset 9
```

Comment:
Can we have a unit test for string casting, we only need this unit test for opset 9

Reply:
Discussed offline, added TODO for unit test as cast operator is not yet supported in Keras->ONNX.